src/internal_events/common.rs

```diff
@@ -111,24 +110,37 @@ impl InternalEvent for CollectionCompleted {
     }
 }
 
+pub const _INTENTIONAL: bool = true;
+pub const UNINTENTIONAL: bool = false;
+
 #[derive(Debug)]
-pub struct ComponentEventsDropped {
+pub struct ComponentEventsDropped<const INTENTIONAL: bool> {
     pub count: u64,
-    pub intentional: bool,
     pub reason: &'static str,
 }
 
-impl InternalEvent for ComponentEventsDropped {
+impl<const INTENTIONAL: bool> InternalEvent for ComponentEventsDropped<INTENTIONAL> {
     fn emit(self) {
-        error!(
-            message = "Events dropped.",
-            intentional = self.intentional,
-            reason = self.reason,
-        );
+        let message = "Events dropped";
+        if INTENTIONAL {
+            debug!(
+                message,
+                intentional = INTENTIONAL,
+                reason = self.reason,
+                internal_log_rate_secs = 10,
+            );
```

Comment:
Could this code path be added later when `INTENTIONAL` could actually be true somewhere?

Reply:
I think @pablosichert  has changes in a branch / PR where `INTENTIONAL` will be true.